docs/custom-flows/error-handling.mdx

```diff
@@ -13,31 +13,69 @@ A user is attempting to sign up with your application, but they are attempting t
 
 <CodeBlockTabs type="framework" options={["React", "JavaScript"]}>
 ```jsx
-import { useState } from "react";
-import { useSignUp } from "@clerk/clerk-react";
+import { useState, SyntheticEvent } from "react";
+import { useSignUp, isClerkAPIResponseError } from "@clerk/clerk-react";
+import { ClerkAPIError } from '@clerk/types';
 
 function Example() {
-  const [error, setError] = useState(null);
-
-  const { signUp } = useSignUp();
-
-  const signUpUser = async () => {
-    const response = await signUp
-      ?.create({
-        identifier: "example@email.com",
-        password: "Tester01!",
-      })
-      .then((res) => console.log("response", res))
-      .catch((err) => setError(err.errors[0].message));
+  const [emailAddress, setEmailAddress] = useState('');
+  const [password, setPassword] = useState('');
+  const [errors, setErrors] = useState<ClerkAPIError[]>();
+
+  const { isLoaded, signUp, setActive } = useSignUp();
+
+  const attemptSignUp = async (e: SyntheticEvent) => {
+    e.preventDefault();
+
+    // clear any errors that may have occured during previous form submission
+    setErrors(undefined);
+
+    if (!isLoaded || !signUp) return;
+
+    try {
+      const result = await signUp.create({ emailAddress, password});
+      if (result.status == 'complete') {
+        await setActive({ session: result.createdSessionId });
+      } else {
+        // handle incomplete sign-up
```

Comment:
Let's add error handling in here as well, using the same logic as line 44

Reply:
Any error thrown by `setActive` is already caught by the existing catch clause.

Could you clarify what extra error handling is needed? Are errors thrown by setActive of a different type, such as ClerkRuntimeError, and thus need to be handled differently?

Reply:
The catch will handles an error, but not `result.status === 'missing_requirements' | 'abandoned'`. See https://clerk.com/docs/references/react/use-sign-up#result-status

This is a non-error return that still should be read. The `abandoned` is something users can run into if they take too long to try and confirm signUp, depending on signUp method. `missing_requirements` will *usually* be found and resolved during dev, but in case the app setting change its good to show versus a silent fail with no message.

Reply:
Got it. These successful errors do not conform to the type `ClerkAPIError` and cannot be handled immediately by the same logic as line 44. Here are two approaches:

1. Create an object that conforms to ClerkAPIError, set the error and it is shown in the UI.
```typescript
if (result.status == 'missing_requirements') {
        const error = { code: result.status, message: 'Requirement from the email, phone, username settings is missing.' }
        setErrors([error]);
        return;
}
```

2. Throw a standard Error object and show it in the console.
```typescript
    if (result.status == 'missing_requirements') {
        throw new Error(result.status);
        return;
    }
} catch(err: unknown) {
    if (isClerkAPIResponseError(err)) {
        setErrors(err.errors);
        return;
    }
    if (err instanceof Error) { 
        console.error(err.message) 
        return;
    }
}
```

Do you have a preference between these two, or is there another approach you'd prefer?

Reply:
Definitely 1 or something like it. The goal with this is two fold. 1. Show the kinds of errors that need to be dealt with and 2. Give an example of showing those to the user.

If a dev wants to handle these different they can definitely can -- this gives them the info what what to handle.